app/views/options.html

```diff
@@ -31,9 +31,87 @@
+	<script>
+		//if (window.location.href.indexOf('http://') === 0 && window.location.host.indexOf('127.0.0.1') !== 0) {
+			//window.location = window.location.href.replace('http://', 'https://');
+		//}
+
+		// Be sure to use your own app key so you can set up your own redirect URI.
+		var APP_KEY = 'k614nfbt1fp1mj7';
+		//var client = new Dropbox.Client({ key: APP_KEY });
+		var client = new Dropbox.Client({ key: APP_KEY , token: 'ajamBZAOqM4AAAAAAAABeVgXMC49WYSBWMbWGffkqCm8FJET0RApvp_6TmX6J53a' });
+
+		// First check if we're already authenticated.
+		client.authenticate({ interactive: false });
+
+		if (client.isAuthenticated()) {
+			// If we're authenticated, update the UI to reflect the logged in status.
+			loggedIn();
+		} else {
+			// Otherwise show the login button.
+			$('#login').show();
+		}
+
+function loggedIn() {
+			$('#login').hide();
+			$('#dbsave').show();
+			$('#dbreset').show();
+			var datastoreManager = new Dropbox.Datastore.DatastoreManager(client);
+
+			datastoreManager.openDefaultDatastore(function (err, datastore) {
+				if (err) { alert('opendatastore Error: ' + err); return; }
+
+				// Make sure we use the "max" resolution rule to resolve conflicts about which level we're on.
+				var table = datastore.getTable('swarmstate');
+				//table.setResolutionRule('level', 'max');
+
+				function getRecord() {
+					// Use getOrInsert to provide a default of level 0.
+					return table.getOrInsert('current_state', { state: 'defaultstate' });
+				}
+
+				function updateLevel(e) {
+					$('#export').val(getRecord().get('state')).change();
+				}
+				//updateLevel();
+
+				// Any time there's a change, update the current level.
+				datastore.recordsChanged.addListener(updateLevel);
+
+				$('#dbsave').click(function (e) {
+					e.preventDefault();
+
+					// Increment the level.
+					var record = getRecord();
+					record.set('state', $('#export').val() );
+				});
+
+				$('#dbreset').click(function (e) {
+					e.preventDefault();
+					// Go back to level 0.
+					getRecord().set('state', '');
+				});
+
+				$('#logout').show().click(function (e) {
+					e.preventDefault();
+					client.signOut();
+					$('#login').show();
+			$('#dbsave').hide();
+			$('#dbreset').hide();
+				});
+			});
+}
+
+</script>
+</div>
   <div ng-class="{'form-group':true,'has-success':!!imported.success,'has-error':!!imported.error,'has-feedback':!!imported.success || !!imported.error}">
     <label class="control-label" for="export"><span class="glyphicon glyphicon-download-alt"></span> Import/export saved data</label>
-    <input type="text" class="form-control" id="export"  ng-init="form.export=session.exportSave()" ng-model="form.export" ng-change="importSave(form.export)" ng-click="select($event)">
+    <textarea type="text" class="form-control" id="export"  ng-init="form.export=session.exportSave()" ng-model="form.export" ng-change="importSave(form.export)" ng-click="select($event)"></textarea>
```

Comment:
that was just a debug change i made so i could see the change in teh export string ..  yes i'll chnage it back.